src/Spd.Resource.Repository/ControllingMemberCrcApplication/ControllingMemberCrcRepository.cs

```diff
@@ -41,8 +41,10 @@ public async Task<ControllingMemberCrcApplicationCmdResp> CreateControllingMembe
         //check contact duplicate
         contact? contact = SharedRepositoryFuncs.GetDuplicateContact(_context, _mapper.Map<contact>(cmd), ct);
         //create or update contact
+        contact? contactToCreate = _mapper.Map<contact>(cmd);
+        contactToCreate.contactid = Guid.NewGuid();
         contact = contact == null ? 
-            await _context.CreateContact(_mapper.Map<contact>(cmd), null, _mapper.Map<IEnumerable<spd_alias>>(cmd.Aliases), ct) : 
```

Comment:
you can do it this way. But if GetDuplicateContact is not null, the code still run mapping to get contactToCreate, and it is not used.
could you optimize the code?

Reply:
better to add in the mapping, but I am not sure if it is feasible.